cpp/models/abm/model.h

```diff
@@ -622,6 +596,32 @@ class Model
                            m_contact_exposure_rates_cache[person.get_location().get()], t, dt, parameters);
     }
 
+    /**
+     * @brief Implementation of Model::get_person. 
+     * This function needs to use a template as otherwise std::is_const would be false.
+     * @param[in] m Model in which the person_id is searched for.
+     * @param[in] person_id A Person's PersonId.
+     * @return A reference to the Person.
+     */
+    template <class M>
+    static std::conditional_t<std::is_const_v<M>, const Person&, Person&> get_person_impl(M& m, PersonId person_id)
+    {
+        if (m.m_person_ids_equal_index) {
+            return m.m_persons[static_cast<uint32_t>(person_id.get())];
+        }
+        else {
+            mio::log_warning("get_person is accessed by PersonId which does not align with the index of the person due "
+                             "to former removal of persons. Therefore m_persons is searched.");
+            auto it = std::find_if(m.m_persons.begin(), m.m_persons.end(), [person_id](auto& person) {
+                return person.get_id() == person_id;
+            });
+            if (it == m.m_persons.end()) {
+                log_error("Given Person is not in this Model.");
+            }
+            return *it;
```

Comment:
This is bad, if itr points to end(), i.e. one position *after* m_persons, you still return *itr, i.e. a Person& to nothing - at least no valid storage. This code should fail after the warning.

Reply:
Yes. I pretty sure we can't recover if the requested ID is not available during simulation. 
Maybe it could be nice to have a function to test whether an ID is available for the setup, but we don't need it right now.